no-std-libc-demo/packages/demo/examples/client.rs

```diff
@@ -0,0 +1,327 @@
+#![no_main]
+#![no_std]
+
+extern crate alloc;
+
+use alloc::borrow::Cow;
+use alloc::format;
+use alloc::string::{String, ToString};
+use alloc::sync::Arc;
+use alloc::vec::Vec;
+use core::str::{self, Utf8Error};
+use pki_types::{DnsName, InvalidDnsNameError};
+
+use ministd::io::{self, Read, Stream, Write};
+use ministd::net::{TcpStream, ToSocketAddrs};
+use ministd::{dbg, entry, eprintln};
+use rustls::client::{ClientConnectionData, UnbufferedClientConnection};
+#[allow(unused_imports)]
+use rustls::version::{TLS12, TLS13};
+use rustls::{
+    AppDataRecord, ClientConfig, ConnectionState, EncodeError, EncryptError, InsufficientSizeError,
+    MayEncryptAppData, RootCertStore, UnbufferedStatus,
+};
+use rustls_provider_example::PROVIDER as CRYPTO_PROVIDER;
+
+const SERVER_NAME: &str = "www.rust-lang.org";
+// const SERVER_NAME: &str = "doc.rust-lang.org";
+const PORT: u16 = 443;
+const MAX_ITERATIONS: usize = 30;
+
+entry!(main);
+
+fn main() -> Result<(), Error> {
```

Comment:
Leaving a note that it would be helpful once https://github.com/rustls/rustls/pull/1583 lands to do a consistency pass between this example (and the later server one) vs the ones that landed in 1583. I think there are a few updates from feedback that would apply here as well.